tests/test_notifications.py

```diff
@@ -0,0 +1,53 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+from selenium.common.exceptions import TimeoutException
+
+from foxpuppet.browser.notifications import (
+    BaseNotification,
+    AddOnInstallBlocked)
+
+
+@pytest.fixture
+def firefox_options(firefox_options):
+    firefox_options.add_argument('-foreground')
+    return firefox_options
+
+
+@pytest.fixture
+def notification(firefox, webserver, selenium):
+    selenium.get(webserver.url())
+    selenium.find_element_by_link_text('restartless_addon_signed.xpi').click()
+    return firefox.wait_for_notification(AddOnInstallBlocked)
```

Comment:
This isn't really a callback, but we're removing the context manager in the new window code in pull #38.